src/lib/operators/index_scan.hpp

```diff
@@ -21,13 +20,13 @@ class AbstractTask;
  */
 class IndexScan : public AbstractReadOnlyOperator {
  public:
-  IndexScan(const std::shared_ptr<const AbstractOperator>& input_operator, const ChunkIndexType index_type,
-            const std::vector<ColumnID>& left_column_ids, const PredicateCondition predicate_condition,
-            const std::vector<AllTypeVariant>& right_values, const std::vector<AllTypeVariant>& right_values2 = {});
+  IndexScan(const std::shared_ptr<const AbstractOperator>& input_operator, const ColumnID indexed_column_id,
+            const PredicateCondition predicate_condition, const AllTypeVariant scan_value);
 
   const std::string& name() const final;
 
-  // If set, only the specified chunks will be scanned. See TableScan::excluded_chunk_ids for usage.
+  // Must be set because only the specified chunks will be scanned. See TableScan::excluded_chunk_ids for usage.
```

Comment:
The IndexScan will work when they are not set, but it will not output anything.
There could be cases where this happens, but there shouldn't be an IndexScan then.